src/views/shared/account-list/account-list.view.tsx

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import clsx from "clsx";
+
+import useAccountListStyles from "src/views/shared/account-list/account-list.styles";
+import AccountView from "src/views/shared/account/account.view";
+// domain
+import { Account, PendingDeposit, CoordinatorState, isHermezAccount } from "src/domain/hermez";
+
+interface AccountListProps {
+  accounts: Account[];
+  preferredCurrency: string;
+  pendingDeposits?: PendingDeposit[];
+  coordinatorState?: CoordinatorState;
+  disabledTokenIds?: number[];
+  onAccountClick: (account: Account) => void;
+}
+
+function AccountList({
+  accounts,
+  preferredCurrency,
+  pendingDeposits,
+  coordinatorState,
+  disabledTokenIds,
+  onAccountClick,
+}: AccountListProps): JSX.Element {
+  const classes = useAccountListStyles();
+
+  function getAccountPendingDeposit(account: Account): PendingDeposit | undefined {
+    return !isHermezAccount(account)
+      ? undefined
+      : pendingDeposits?.find((deposit) => deposit.accountIndex === account.accountIndex);
+  }
+
+  function hasAccountPendingDeposit(account: Account): boolean {
+    return pendingDeposits !== undefined && getAccountPendingDeposit(account) !== undefined;
+  }
+
+  function isAccountDisabled(account: Account): boolean {
+    return disabledTokenIds?.find((id) => account.token.id === id) !== undefined;
+  }
+
+  return (
+    <div className={classes.root}>
+      {accounts.map((account, index) => {
+        return (
+          <div
+            key={(isHermezAccount(account) && account.accountIndex) || account.token.id}
+            className={clsx({ [classes.accountSpacer]: index > 0 })}
+          >
+            <AccountView
+              balance={account.balance}
+              preferredCurrency={preferredCurrency}
+              fiatBalance={account.fiatBalance || 0}
```

Comment:
I'm not yet sure about this. Having a fiat balance of zero is not the same as having a fiat balance `undefined`. While here it may be ok showing a zero, somewhere else `undefined` may end up being an error or requiring a different behavior.

If we "fake" a zero in persistence then there's no way for us to make a later decision in the lower layers.

Does this make sense?

Reply:
Mmmm makes sense yep, I totally agree!